gcm_filters/kernels.py

```diff
@@ -121,8 +121,20 @@ class IrregularCartesianLaplacianWithLandMask(BaseLaplacian):
     def __post_init__(self):
         np = get_array_module(self.wet_mask)
 
-        self.w_wet_mask = self.wet_mask * np.roll(self.wet_mask, -1, axis=-1)
-        self.s_wet_mask = self.wet_mask * np.roll(self.wet_mask, -1, axis=-2)
+        err_w = np.where(self.kappa_w > 1.)[0]
+        if err_w.size > 0 :
+            raise ValueError(f"There are kappa_w values > 1 and this can cause the filter to blow up."
+                             f"Please make sure all kappa_w are <=1.")
+
+        err_s = np.where(self.kappa_s > 1.)[0]
+        if err_s.size > 0:
+            raise ValueError(f"There are kappa_s values > 1 and this can cause the filter to blow up."
+                             f"Please make sure all kappa_s are <=1.")
```

Comment:
Thanks for adding this! 

It would be nice to "test this test"; that is, we could add a test to `test_kernels.py` which checks that the error gets actually triggered, if the user puts in `kappa > 1`. I imagine the test could look almost identical to the test I wrote here (in my test I'm checking that an error gets triggered for the `TRIPOLAR` Laplacians if the wet mask has unexpected values):

https://github.com/NoraLoose/gcm-filters/blob/0591b7aae5045e1f90363d5229dec68563b7e0ac/tests/test_kernels.py#L72-L84

Does this make sense?

Reply:
I know I am late to this PR, but I will weigh in quickly with my opinion.

I don't think we should do these kinds of checks. They may slow things down considerably, especially for large arrays. And where do we stop? Should we also check `area`?

 Instead, we should document what's required (positive values) and trust the user to read it.

Reply:
I suggested this because the check would be part of the \__post_init\__ method. Since the \__post_init\__ method is ideally only called once (I know the realization of this is still on our todo list) and the check acts on one 2D array only, I was hoping it wouldn't slow things down by very much. But I'm okay with not doing these kind of checks, and instead documenting everything carefully.

Reply:
> Since the __post_init__ method is ideally only called once (I know the realization of this is still on our todo list)

Yes that would be ideal! But for now that it not how it works...it looks like `__post_init__` is unfortunately getting called on every chunk. 😢

Reply:
@rabernat and @NoraLoose, Thanks for the feedback. I will convert this PR to draft until we decide on the right approach.

Reply:
I have done some profiling and convinced myself that the `__post_init__` cost is not significant in `IrregularCartesianLaplacianWithLandMask`. So I have no problem with these checks.

Later down the road we can do some more serious profiling and try to optimize performance further. But doing that now is premature.

Sorry for slowing things down. @gustavo-marques you can mark as ready for review again and I'll merge.

Reply:
There is already a simple test for this in `test_filter.py`.  Should I add something else?

Reply:
Yes, as Nora said above, you need to test the if block you have introduced. Currently the code that checks for the kappas < 1 is untested. You should use `pytest.raises` (as in https://github.com/NoraLoose/gcm-filters/blob/0591b7aae5045e1f90363d5229dec68563b7e0ac/tests/test_kernels.py#L83-L84) to trigger the error in the test function.